chia/data_layer/data_store.py

```diff
@@ -655,3 +761,53 @@ async def get_proof_of_inclusion_by_key(
         async with self.db_wrapper.locked_transaction(lock=lock):
             node = await self.get_node_by_key(key=key, tree_id=tree_id, lock=False)
             return await self.get_proof_of_inclusion_by_hash(node_hash=node.hash, tree_id=tree_id, lock=False)
+
+    async def get_tree_diff_terminal_nodes(
+        self,
+        tree_id: bytes32,
+        hash: Optional[bytes32],
+        generation_1: int,
+        generation_2: int,
+        *,
+        lock: bool = True,
+    ) -> List[TerminalNode]:
+        async with self.db_wrapper.locked_transaction(lock=lock):
+            if hash is None:
+                return []
+            result: List[TerminalNode] = []
+            queue: List[bytes32] = []
+            queue.append(hash)
+            while len(queue) > 0:
+                node_hash = queue.pop()
+                if not await self.node_exists(node_hash, tree_id, generation_2, lock=False):
+                    node = await self.get_node(node_hash, lock=False)
+                    if isinstance(node, TerminalNode):
+                        result.append(node)
+                    else:
+                        queue.append(node.left_hash)
+                        queue.append(node.right_hash)
+
+            return result
+
+    async def get_tree_diff(
+        self,
+        tree_id: bytes32,
+        root_1: Root,
+        root_2: Root,
+        *,
+        lock: bool = True,
+    ) -> List[DiffData]:
```

Comment:
Since there's no meaning to the order and there shouldn't be duplicates, would a `set` be more appropriate?